src/compiler/scanner.ts

```diff
@@ -1376,6 +1376,9 @@ namespace ts {
                 if (ch >= CharacterCodes.a && ch <= CharacterCodes.z) {
                     const keyword = textToKeyword.get(tokenValue);
                     if (keyword !== undefined) {
+                        if (tokenFlags & TokenFlags.ExtendedUnicodeEscape || "" + tokenValue !== text.substring(tokenPos, pos)) {
+                            error(Diagnostics.Keyword_must_not_contain_escaped_characters, tokenPos, pos - tokenPos);

```

Comment:
unfortunately this is not the correct position (otherwise I would have already submitted a PR for this). You need to consider contextual keywords (`await` and `yield` and probably TypeScript's type keywords): `\u0061wait` is allowed in non-`async` function, but disallowed in `async` functions

Reply:
solved~